### PR TITLE
Fix invalidation of bad caches. Fixes #1929

### LINKED ISSUE #1929: Invalidate Bad Caches "File not found"
An issue to address the user report: http://gatkforums.broadinstitute.org/gatk/discussion/8873/file-not-found-errors-when-using-call-caching/

### PR DESCRIPTION
**Pull request by cjllanwarne**
Fixes Cromwell so that we can add in https://github.com/broadinstitute/centaur/pull/145

### CODE REVIEW

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala [hunk 1]
```diff
@@ -129,7 +129,7 @@ class EngineJobExecutionActor(replyTo: ActorRef,
       log.debug("Cache miss for job {}", jobTag)
       runJob(data)
     case Event(hit: CacheHit, data: ResponsePendingData) =>
-      fetchCachedResults(jobDescriptorKey.call.task.outputs, hit.cacheResultIds.head, data)
+      fetchCachedResults(jobDescriptorKey.call.task.outputs, hit.cacheResultIds.head, data.copy(cacheHit = Option(hit)))
```
Comment (mcovarr) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala]: This doesn't seem like it's fixing the test so much as the production code?

**Comment by mcovarr** [comment]
:+1:

**Comment by ruchim** [comment]
:+1:

File: .travis.yml [hunk 2]
```diff
@@ -17,7 +17,7 @@ before_cache:
   - find $HOME/.sbt -name "*.lock" -delete
 env:
   global:
-    - CENTAUR_BRANCH=develop
+    - CENTAUR_BRANCH=cjl_invalidate_bad_cache_test
```
Comment (cjllanwarne) [inline .travis.yml]: Update before merging

**Pull request merged**